pycalphad/tests/test_calculate.py

```diff
@@ -106,3 +108,17 @@ def test_single_model_instance_raises():
     mod = Model(DBF, comps, 'L12_FCC')  # Model instance does not match the phase
     with pytest.raises(ValueError):
         calculate(DBF, comps, ['LIQUID', 'L12_FCC'], T=1400.0, output='_fail_', model=mod)
+
+
+def test_missing_phase_records_passed_to_calculate_raises():
+    "calculate should raise an error if all the active phases are not included in the phase_records"
+    my_phases = ['LIQUID', 'FCC_A1']
+    subset_phases = ['FCC_A1']
+    comps = ['AL', 'FE', 'VA']
+
+    models = instantiate_models(ALFE_DBF, comps, subset_phases)
+    # Dummy conditions are just needed to get all the state variables into the PhaseRecord objects
+    phase_records = build_phase_records(ALFE_DBF, comps, subset_phases, {v.N: None, v.P: None, v.T: None}, models)
```

Comment:
`{v.StateVariable: None, ...}` makes me think the function signature of `build_phase_records` should change to accept the state variables as a list or set instead. The dict could be kept for backwards compatibility (temporarily) since it's just 1-2 lines to extract the keys. Or do we want the flexibility to backtrack on multi-phase conditions down the line?

Reply:
I'm in favor of accepting a sequence of state variables here. Our next release also changes some of the internal solver APIs, so we might as well feel free to change others as well.

Reply:
I changed the name of the `conds` argument of `build_phase_records` to `state_variables`, so that covers the semantic change.

For the implementation, I made a [slight modification](https://github.com/pycalphad/pycalphad/pull/361/commits/854cf4e4951e7596d1f6a20d2bdc55253ebb6ca8#diff-47b2bfb2462e33b519b7489e73ff5bac523728463855804c4d40ea7ced4bb26e) to `get_state_variables` to loop over `conds` directly instead of accessing `conds.keys()`. With this modification, we can take advantage of the fact that iterating over Python dictionaries means iterating over the keys. Thus, passing dictionaries of conditions to `state_variables` is backwards compatible and passing any iterable (list, set, etc.) of state variables also works.